distributed/shuffle/_shuffle_extension.py

```diff
@@ -590,12 +667,39 @@ def get(
             self.completed_workers[id] = set()
 
         return {
+            "status": "OK",
             "worker_for": self.worker_for[id],
             "column": self.columns[id],
             "schema": self.schemas[id],
             "output_workers": self.output_workers[id],
         }
 
+    async def remove_worker(self, scheduler: Scheduler, worker: str) -> None:
+        broadcasts = []
+        for shuffle_id, output_workers in self.output_workers.items():
+            if worker not in output_workers:
+                continue
+            self.erred_shuffles[shuffle_id] = worker
+            contact_workers = output_workers.copy()
+            contact_workers.discard(worker)
+            message = f"Worker {worker} left during active shuffle {shuffle_id}"
+            broadcasts.append(
+                scheduler.broadcast(
+                    msg={
+                        "op": "shuffle_set_exception",
+                        "message": message,
+                        "shuffle_id": shuffle_id,
+                    },
+                    workers=list(contact_workers),
```

Comment:
I've adjusted this to all participating workers, i.e., output workers as well as any worker that ever asked the scheduler about the shuffle (and should therefore be an input worker). This should allow us to fail earlier. Not messaging workers that have not participated until `remove_worker` is called should be fine since those will call `shuffle_get` _after_ the shuffle has been erred on the scheduler, returning an error.